# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/generic/nsBlockFrame.cpp
# Commit: dd14514b9e99
# Full Hash: dd14514b9e992d986c6802c434e59d67a172f828
# Author: Ting-Yu Lin <tlin@mozilla.com>
# Date: 2022-10-12 09:45:09
# Regressor Bug: 1794456
# File Overlap Count: 1
# Description:
#   Bug 1794456 Part 3 - Support range-based for loop for nsFrameList::Slice. r=emilio
#   
#   Since Slice::mEnd is the first frame that is NOT in the slice, so we only
#   support forward iteration, and it's sufficient to replace existing usages of
#   nsFrameList::Enumerator.
# ==============================================================================

diff -r 58733fee05e0 -r dd14514b9e99 layout/generic/nsBlockFrame.cpp
--- a/layout/generic/nsBlockFrame.cpp	Tue Oct 11 21:01:27 2022 +0000
+++ b/layout/generic/nsBlockFrame.cpp	Tue Oct 11 21:01:28 2022 +0000
@@ -5953,8 +5953,7 @@
 
   // Walk through the new frames being added and update the line data
   // structures to fit.
-  for (nsFrameList::Enumerator e(newFrames); !e.AtEnd(); e.Next()) {
-    nsIFrame* newFrame = e.get();
+  for (nsIFrame* newFrame : newFrames) {
     NS_ASSERTION(!aPrevSibling || aPrevSibling->GetNextSibling() == newFrame,
                  "Unexpected aPrevSibling");
     NS_ASSERTION(
@@ -8166,8 +8165,8 @@
   // Pushed floats must not have a next-in-flow in mFloats or mFrames.
   oofs = GetPushedFloats();
   if (oofs) {
-    for (nsFrameList::Enumerator e(*oofs); !e.AtEnd(); e.Next()) {
-      nsIFrame* nif = e.get()->GetNextInFlow();
+    for (nsIFrame* f : *oofs) {
+      nsIFrame* nif = f->GetNextInFlow();
       MOZ_ASSERT(!nif ||
                  (!mFloats.ContainsFrame(nif) && !mFrames.ContainsFrame(nif)));
     }